c/cmake_modules/AdbcDefines.cmake

```diff
@@ -92,6 +92,17 @@ elseif(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang"
       -Werror
       -Wno-unused-parameter)
   set(ADBC_C_CXX_FLAGS_PRODUCTION -Wall)
+
+  string(APPEND C_DEBUG_FLAGS " -Og -ggdb")
+  string(APPEND CXX_DEBUG_FLAGS " -Og -ggdb")
+  string(APPEND C_RELWITHDEBINFO_FLAGS " -Og -ggdb")
+  string(APPEND CXX_RELWITHDEBINFO_FLAGS " -Og -ggdb")
```

Comment:
How about removing `-Og` here because `RelWithDebInfo` is one of optimization builds?